wordpress-mobile/WordPress-iOS
# Support failed image reloading on Reader posts

To improve the user experience on the Reader screen when images fail to download, [issue 12517 was created](https://github.com/wordpress-mobile/WordPress-iOS/issues/12517) to display a retry button allowing the user to retry the download. Currently, if the image is downloaded successfully, this area is occupied by the image. If the image fails to download for any reason, a blank gray view occupies this area. The ask here is to replace this view with a tappable retry button.

### Demo

| **Before**  | **After** |
| ------------- | ------------- |
| ![original](https://user-images.githubusercontent.com/1898325/66795614-864db980-eedb-11e9-9211-d891d7a27b52.gif) | ![demo](https://user-images.githubusercontent.com/1898325/66795243-69fd4d00-eeda-11e9-8721-49845aaddf76.gif) |

### Testing 

1. Log in with valid credentials and tap the Reader tab
2. Navigate back to the Reader home screen and tap the Search option
3. Search for the following post: https://alexdanco.com/
4. If testing on a physical device, activate the 100% Loss option in the Network link Conditioner (located in Settings)
5. If testing on a simulator, download and activate the Apple Network Link Conditioner to the 100% Loss profile
6. Tap a post that contains images (e.g. "Why I don’t love Light Rail transit" - https://alexdanco.com/2019/09/12/why-i-dont-love-light-rail-transit/)
7. Observe that when the timeout completes, the retry view is now shown in the place of an image that failed to download
8. If an internet connection is reenabled, images should download, as before
9. If an image is tapped, it should either be displayed as a preview or open an in-app browser, as before

### Known Issues

1. SVG images are not supported: https://github.com/wordpress-mobile/WordPress-iOS/pull/12634#issuecomment-541345870
2. Issues covered here: https://github.com/wordpress-mobile/WordPress-iOS/pull/12634#issuecomment-542001552

### Pending

- [ ] Also see the summary of pending issues in https://github.com/wordpress-mobile/WordPress-iOS/pull/12634#issuecomment-544165492
- [ ] Test Reader comments for regressions [1] (see below)
- [ ] Handle cases where image area is too small to fit the retry view
- [x] Check retry view font size

[1] - This PR modifies `WPRichTextImage` which is also used on the reader comments screen, so there is a potential for errors to be introduced here. I have been unable to test so far due to not having a blog with comment attachments enabled readily available.
